### PR TITLE
Add FedEx.  Add carrier_description.

### PR DESCRIPTION
**Pull request by namotco**
Please look over the Carp part:

-our @CARP_NOT = ('Webservice::Shipment'); # don't carp from AUTOLOAD

I guess that line got removed in my code, but I don't understand why.

### CODE REVIEW

File: lib/Webservice/Shipment/Carrier.pm [hunk 1]
```diff
@@ -7,13 +7,13 @@ use Mojo::UserAgent;
...
 
 use Carp;
-our @CARP_NOT = ('Webservice::Shipment'); # don't carp from AUTOLOAD
 
 has api_url => sub { Mojo::URL->new };
 has password => sub { croak 'password is required' };
 has ua => sub { Mojo::UserAgent->new };
 has username => sub { croak 'username is required' };
 has [qw/date_format validation_regex/];
+has carrier_description => sub { croak 'carrier_description is required' };
```
Comment (jberger) [inline lib/Webservice/Shipment/Carrier.pm]: This shouldn't be a required attribute. In fact it probably shouldn't be overridable should it? `sub carrier_description { undef }` which gets overloaded like `sub carrier_description { 'FedEx' }`. Would anyone ever want/need to change this value at run-time?

**Comment by jberger** [comment]
The style is rather inconsistent with the existing code. Also no tests were added for the FedEx service.

**Comment by namotco** [comment]
Thanks for the feedback.  I’ll work on the tests and style.

You’re correct you would not want to override the carrier_description.  I do think the carrier_description should be required though, but really I just wanted to be able to do something like

say $package->carrier;

Maybe there’s a better way?

**Comment by jberger** [comment]
Closing in favor of #4

**Pull request closed without merging**